### PR TITLE
Refactor/pm 1058/saving id instead of name

### PR DESCRIPTION
**Pull request by mati-1**
[PM-1058](https://playmakerpro.atlassian.net/browse/PM-1058)

### Task Description

- added updating user data modal
- added autosubmitting in login page by code when user types/pastes code
- added missing data in the base data section
- added replacing bolded letters by search value
- changed logic for selecting item by id
- fixed responsiveness in the register and login pages
- removed related to incorrect email errors on login page

### CODE REVIEW

File: components/login-register/register/pages/stats-page/stats-page.tsx [hunk 1]
```diff
@@ -89,7 +90,7 @@ export const StatsPage = () => {
               isFirstValueIncorrect ? 'Podaj poprawny 90 minut URL' : ''
             }
             inputValue={firstUrlValue}
-            onChange={event => setFirstUrlValue(event.target.value)}
+            onChange={({ target: { value } }) => setFirstUrlValue(value)}
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/pages/stats-page/stats-page.tsx:93]: hmm, now it looks much better imo

File: components/login-register/common/Input/toggleSwitch.tsx
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/common/Input/toggleSwitch.tsx:1]: oh my gooood 😿

File: components/login-register/register/pages/positions-page/positions-page.tsx [hunk 2]
```diff
@@ -25,52 +26,44 @@ export const PositionsPage = () => {
...
-  )
-  const [innerSelectedMainPos, setInnerSelectedMainPos] = useState(
-    ctxMainPosition || [''],
-  )
-  const [innerSelectedAltPos, setInnerSelectedAltPos] = useState<string[]>(
-    ctxAltPosition || [],
-  )
   const [openMainModal, setOpenMainModal] = useState(false)
   const [openAltModal, setOpenAltModal] = useState(false)
+  const { data: positionsList, isLoading } = usePositionsList(openMainModal)
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/pages/positions-page/positions-page.tsx:31]: good

File: components/login-register/register/register-context.tsx [hunk 3]
```diff
@@ -41,18 +41,18 @@ const initialRegisterState: IRegisterState = {
...
-  city: [''],
-  mainPosition: [''],
+  citizenship: [],
+  languages: [],
+  city: [],
+  mainPosition: [],
   altPosition: [],
   betterLeg: 'right',
-  team: initialClubState,
+  team: '',
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/register-context.tsx:55]: good point

File: components/profile/overview/baseData/helpers.ts [hunk 4]
```diff
@@ -27,9 +28,10 @@ export const getBaseData = ({
   status,
   languages,
   citizenships,
+  born,
 }: IProps) => {
   const baseData = {
-    Wiek: `${age} lat (${new Date().getFullYear() - age})`,
+    Wiek: `${age} lat (${new Date(born).getFullYear()})`,
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/profile/overview/baseData/helpers.ts:34]: why is this incorrect? 🤔

File: components/login-register/register/pages/positions-page/helpers.tsx [hunk 5]
```diff
@@ -11,29 +11,29 @@ import RightDefender from '@/icons/register/RightDefenderField.svg'
...
+    case '8':
       return <LeftMidfielder className={clsx(styles, className)} />
-    case 'Prawy pomocnik':
+    case '9':
       return <RightMidfielder className={clsx(styles, className)} />
-    case 'Napastnik':
+    case '10':
       return <Striker className={clsx(styles, className)} />
     default:
       return <GoalKeeper className={clsx(styles, className)} />
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/pages/positions-page/helpers.tsx:39]: yeah, i'd forgotten bout that

File: components/login-register/common/Loading/modal-loader.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+export const ModalLoader = () => (
+  <motion.div
+    initial={{ opacity: 1 }}
+    animate={{ opacity: 0 }}
+    exit={{ opacity: 0 }}
+    className="flex h-full w-full items-center justify-center"
+  >
+    <Spinner size="medium" desktopSize="large" />
+  </motion.div>
+)
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/common/Loading/modal-loader.tsx:14]: good point

File: components/login-register/common/Input/toggleSwitch.tsx [hunk 7]
```diff
@@ -52,9 +52,11 @@ export const ToggleSwitch = ({
...
   const handleToggleValue = () => {
-    const newValue = currentValue === firstValue ? secondValue : firstValue
-    setCurrentValue(newValue)
-    setCtxValue(newValue)
+    setTimeout(() => {
+      const newValue = currentValue === firstValue ? secondValue : firstValue
+      setCurrentValue(newValue)
+      setCtxValue(newValue)
+    }, 300)
   }
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/common/Input/toggleSwitch.tsx:60]: cos when the user spam the button, the animation is bugged (i think in this case we have to use mf)

File: components/login-register/register/pages/positions-page/helpers.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/pages/positions-page/helpers.tsx:39]: i'll just put position names, for now

File: components/login-register/common/Input/toggleSwitch.tsx [hunk 8]
```diff
@@ -52,9 +52,11 @@ export const ToggleSwitch = ({
   const values = [firstValue, secondValue]
 
   const handleToggleValue = () => {
-    const newValue = currentValue === firstValue ? secondValue : firstValue
-    setCurrentValue(newValue)
-    setCtxValue(newValue)
+    setTimeout(() => {
+      const newValue = currentValue === firstValue ? secondValue : firstValue
+      setCurrentValue(newValue)
+      setCtxValue(newValue)
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/common/Input/toggleSwitch.tsx:58]: indeed

**Review by dejwi — CHANGES_REQUESTED** [review]
move the input field to components and I think we can go with this

**Review by dejwi — APPROVED** [review]

**Pull request merged**